lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1176,26 +1176,27 @@ ChangeResult TypeAnalyzer::visitAtenNllLossForwardOp(
   auto self = operands[0]->getValue();
   auto outputKnowledge =
       ValueKnowledge::getNotNonePessimisticValueState(op.getContext());
-
-  // Contains Knowledge of shape and dtype for the 1st result.
-  outputKnowledge.dtype = self.dtype;
-  int64_t reduction;
-  unsigned resultRank = self.sizes.size();
-
-  // Contains Knowledge of shape and dtype for the 2nd result.
   auto totalWeightKnowledge =
       ValueKnowledge::getNotNonePessimisticValueState(op.getContext());
+
+  // Contains Knowledge of dtype for both results.
+  outputKnowledge.dtype = self.dtype;
   totalWeightKnowledge.dtype = self.dtype;
-  totalWeightKnowledge.sizes.resize(0, kUnknownSize);
-  totalWeightKnowledge.hasSizes = true;
+  if (self.hasSizes) {
+    int64_t reduction;
+    unsigned resultRank = self.sizes.size();
```

Comment:
`resultRank` is only needed in `matchPattern` if block. Please move to inside the block to make the scope more clear.